src/Sentry/Laravel/Tracing/Storage/TracingFilesystem.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Sentry\Laravel\Tracing\Storage;
+
+use Illuminate\Contracts\Filesystem\Cloud as CloudFilesystemContract;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemAdapter;
+use Sentry\Tracing\SpanContext;
+use function Sentry\trace;
+
+class TracingFilesystem implements CloudFilesystemContract
+{
+    protected $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @template TResult
+     * @param array<string, mixed> $data
+     * @param callable(): TResult $originalCall
+     * @return TResult
+     */
+    protected function withTracing(string $method, array $data, callable $originalCall)
+    {
+        $context = new SpanContext();
+        $context->setOp("storage.{$method}");
+        $context->setData($data);
```

Comment:
Should we add a custom `$description` for a better user experience, or is just data sufficient?

Reply:
The filename, if accessible, would be a good description.
You could also construct the data array with prefixed keys, like `file.path` and such.

Reply:
Given that some operations work with directories or two paths, I think I am going to stick to just data to avoid redundant information.